src/ServiceManagement/Common/Commands.ScenarioTest/Resources/ServiceManagement/ServiceManagementTests.ps1

```diff
@@ -296,6 +296,73 @@ function Run-AutoGeneratedVirtualMachineCmdletTests
     }
 }
 
+# Run dSMS Hosted Service test
+function Run-DSMSHostedServiceTest
+{
+	# Setup
+    $svcName = 'pstest' + (Get-CloudServiceName);
+	$location = Get-DefaultLocation;
+
+    $storageName = 'pstest' + (getAssetName);
+    New-AzureStorageAccount -StorageAccountName $storageName -Location $location;
+
+    # Associate the new storage account with the current subscription
+    Set-CurrentStorageAccountName $storageName;
+
+	$testMode = Get-ComputeTestMode;
+    if ($testMode.ToLower() -ne 'playback')
+    {
+        $cspkg = '.\Resources\ServiceManagement\Files\dSMSTest.cspkg';
+    }
+    else
+    {
+        $cspkg = "https://${storageName}.blob.azure.windows.net/blob/dSMSTest.cspkg";
+    }
+    $cscfg = "$TestOutputRoot\Resources\ServiceManagement\Files\dSMSTest.cscfg";
+	$cscfgChanged = "$TestOutputRoot\Resources\ServiceManagement\Files\dSMSTest-changed.cscfg";
+	
+	# Create a temporary self-signed cert
+	$cert = New-SelfSignedCertificate -DnsName "example.local" -CertStoreLocation "Cert:\CurrentUser\My";
+	$certPath = "Cert:\CurrentUser\My\$($cert.Thumbprint)"
+	# Update the cscfg to use the new cert
+	(Get-Content $cscfg) | ForEach-Object {$_ -replace "\{\{PLACEHOLDER\}\}", $cert.Thumbprint} | Set-Content $cscfgChanged;
+
+    try
+    {
+		# Create Hosted Service
+		$result = New-AzureService -ServiceName $svcName -Location $location -Label $svcName -Description $svcName;
+
+		# Upload the certificate
+		Add-AzureCertificate -ServiceName $svcName -CertToDeploy $cert;
+
+        # Deploy to staging
+        $result = New-AzureDeployment -ServiceName $svcName -Package $cspkg -Configuration $cscfgChanged -Label $svcName -Slot Staging;
+
+		# Get Deployment
+		$deploy = Get-AzureDeployment -ServiceName $svcName -Slot Staging;
+
+		# Make a change
+		$newConfig = $deploy.Configuration -replace 'Setting name="DummySetting" value="Foo"', 'Setting name="DummySetting" value="Bar"';
+		$newConfig | Set-Content $cscfgChanged;
+
+		# Update configuration
+		$result = Set-AzureDeployment -Config -ServiceName $svcName -Configuration $cscfgChanged -Slot Staging;
```

Comment:
Why do you need to get a new config and do a Set-AzureDeployment on the new config? I don't quite understand.

Reply:
It's to test round-trip. Deploy (tests New) Get (tests Get) Update (tests Set).
I wanted to make sure all parts could handle config without thumbprints.